var/spack/repos/builtin/packages/ghostscript/package.py

```diff
@@ -29,14 +29,23 @@ class Ghostscript(Package):
     """An interpreter for the PostScript language and for PDF."""
 
     homepage = "http://ghostscript.com/"
-    url = "http://downloads.ghostscript.com/public/old-gs-releases/ghostscript-9.18.tar.gz"
```

Comment:
Your `list_url` worked for me. I didn't expect spack to know how to parse the output, which seems to be a fairly complex html document.

Should I keep the `list_url` in the package for now? Or is your generic solution ready to go?